### ISSUE TITLE
Add LAN to LAN filtering rules

### ISSUE THREAD

**Issue by abelbeck** [arno-iptables-firewall/aif#28]
By default, LAN's are isolated from each other, a good thing for example for a guest WiFi VLAN.

But in special situations it may be desirable to allow certain LAN's access to specific network resources, such as network printers, possibly advertised using mDNS using avahi and 'enable-reflector=yes'.

The problem is currently AIF only supports TRUSTED_IF and IF_TRUSTS beyond the isolated default. All or nothing.

I propose an additional chain LAN_LAN_FORWARD_CHAIN and set of rules LAN_LAN_HOST_OPEN_xxx, modeled after DMZ_LAN_HOST_OPEN_xxx to selectively allow LAN to LAN traffic by src, dst and port.

I propose a new function setup_lan_lan_forward_chain() implementing the new rules which would be called...
```
    # Setup helper chain for the LAN:
    setup_lan_inet_forward_chain;
    setup_lan_lan_forward_chain;  <<< added <<<

    IFS=' ,'
    for iif in $INT_IF; do
      echo "Applying internal(LAN) FORWARD policy to interface: $iif"
      
      # Always make subnets on the SAME interface trust each other
      iptables -A FORWARD -i $iif -o $iif -j ACCEPT

      for eif in $EXT_IF; do
        iptables -A FORWARD -i $iif -o $eif -j LAN_INET_FORWARD_CHAIN
      done

    <<< hook in LAN_LAN_FORWARD_CHAIN using INT_IF permutations >>>
    done
```

@arnova do you agree ?

**Comment by abelbeck** [comment]
I have a commit ready to apply, tested with AstLinux and works as expected.

This seems like something AIF should have for completeness, the only negative is it adds (N * (N - 1)) FORWARD chain entries jumping to the single LAN_LAN_FORWARD_CHAIN, (where N = number of INT_IF entries).  For typical values of N this is still quite reasonable.

While we are looking at this, there has always been two other related chains:
```
INT_FORWARD_IN_CHAIN
INT_FORWARD_OUT_CHAIN
```
which are not used by any AIF rules or plugins, and these adds (2 * N) FORWARD chain entries for the same N above.

There is a risk some user has plugged into INT_FORWARD_*_CHAIN, though they seem unused, we may consider removing these chains in favor of the new LAN_LAN_FORWARD_CHAIN and document it well.  I would think most users would use either the FORWARD_CHAIN or POST_FORWARD_CHAIN to add custom rules.

**Comment by abelbeck** [comment]
Another thought, we could optimize this by only adding the  (N * (N - 1)) LAN_LAN_FORWARD_CHAIN references to the FORWARD chain only when LAN_LAN_HOST_OPEN_xxx rules are defined.

So by default this would be no change.  The LAN_LAN_FORWARD_CHAIN would be private to AIF.

**Comment by arnova** [comment]
@abelbeck : Sorry for my somewhat delayed response. I recently got married and just got back from our honeymoon :-) Your suggestion sounds good. The chains were once implemented for possible use by plugins. Since we can imo only support 3rd party plugins which are submitted back to us, I'm fine with dropping those chains and replacing them with something else if that's more convenient.

**Comment by abelbeck** [comment]
@arnova CONGRATULATIONS !!! Hearty handshake.

I'm happy with the solution I committed 121acf5, we have been using this in AstLinux and seems like it should have always been there.

I left the INT_FORWARD_x_CHAIN's as is, since the LAN_LAN_FORWARD_CHAIN is not added by default.  It would not bother me if the INT_FORWARD_x_CHAIN's were removed, but someone out there may have have plugged into them.

**Comment by arnova** [comment]
Thanks Lonnie, much appreciated!

If one can do exactly the same thing with the new chains, I'd prefer the old ones. The chances of it actually being used somewhere out there are quite slim. We could also add a deprecate warning (by detecting the chain has rules and generating a warning if so), and remove it somewhere in the future ?

**Comment by abelbeck** [comment]
You said "... I'd prefer the old ones."  Did you mean to say "... I'd prefer removing the old ones." ?

People probably don't usually look at the AIF log messages until something doesn't work, so I don't think a deprecation warning is needed, a CHANGELOG notice is more likely to be read.  I would think most users would use either the FORWARD_CHAIN or POST_FORWARD_CHAIN to add custom rules in this case.

You decide, I'll be happy to remove the INT_FORWARD_IN_CHAIN and INT_FORWARD_OUT_CHAIN if you give the go.  Or feel free yourself.

Also probably time to start thinking about a fresh release.

**Comment by arnova** [comment]
@lonnie: Somehow the formatting got screwed up, removing parts of the sentence (happened to me before on Github). But yes, I meant removing the old ones.

I'd say: let's get rid of it then. Btw. I just noticed this new lan-lan functionality to be on my todo-list because I needed for a project at work. Great it's already there now :-)

A new release would be a good idea yes.

**Comment by abelbeck** [comment]
Fixed in 2.0.1g
